pkg/auth/iam/iam.go

```diff
@@ -137,6 +135,24 @@ func (filter *Filter) Auth(opts ...FilterOption) restful.FilterFunction {
 			}
 		}
 
+		if filter.options.SubdomainValidationEnabled {
+			if valid := validateSubdomainAgainstNamespace(getHost(req.Request), claims.Namespace, filter.options.SubdomainValidationExcludedNamespaces); !valid {
+				svcErr := respondError(http.StatusNotFound, SubdomainMismatch, "data not found: "+ErrorCodeMapping[SubdomainMismatch])
```

Comment:
Want to confirm: does the scenario fit for 404?

Reply:
Yeah. We want to return 404 when user call API in the wrong domain. So the error in the website will be something like "The page you requested doesn't exist"